cmd/controller-manager/app/options/options.go

```diff
@@ -102,7 +100,7 @@ func (s *ServerOption) AddFlags(fs *pflag.FlagSet) {
 
 // CheckOptionOrDie checks the LockObjectNamespace.
 func (s *ServerOption) CheckOptionOrDie() error {
-	if s.EnableLeaderElection && s.LockObjectNamespace == "" {
+	if s.LeaderElection.LeaderElect && s.LeaderElection.ResourceLock == "" {
```

Comment:
Here we can call `ValidateLeaderElectionConfiguration` to check all params.

Reply:
All that is updated!